### PR TITLE
[ess_billing] Handle null instances in billing response

### LINKED ISSUE #17470: [Stack 8.19.12-SNAPSHOT] [ess_billing] Failing test daily: system test: normal in ess_billing.billing
- Stack version: 8.19.12-SNAPSHOT
- Package: ess_billing
- Failing test: system test: normal
- DataStream: billing
- Owners:
    - @elastic/customer-architects
    - @elastic/obs-infraobs-integrations

Failure:
```
test case failed: could not find the expected hits in metrics-ess_billing.billing-92774 data stream
```

First build failed: https://buildkite.com/elastic/integrations/builds/38545

Latest 5 failed builds:
- https://buildkite.com/elastic/integrations/builds/40483
- https://buildkite.com/elastic/integrations/builds/40490
- https://buildkite.com/elastic/integrations/builds/40497
- https://buildkite.com/elastic/integrations/builds/40594
- https://buildkite.com/elastic/integrations/builds/40665

### LINKED ISSUE #17596: [Stack 9.4.0-SNAPSHOT] [ess_billing] Failing test daily: system test: normal in ess_billing.billing
- Stack version: 9.4.0-SNAPSHOT
- Package: ess_billing
- Failing test: system test: normal
- DataStream: billing
- Owners:
    - @elastic/customer-architects
    - @elastic/obs-infraobs-integrations

Failure:
```
test case failed: could not find the expected hits in metrics-ess_billing.billing-61739 data stream
```

First build failed: https://buildkite.com/elastic/integrations/builds/38908

Latest 5 failed builds:
- https://buildkite.com/elastic/integrations/builds/40861
- https://buildkite.com/elastic/integrations/builds/40867
- https://buildkite.com/elastic/integrations/builds/40897
- https://buildkite.com/elastic/integrations/builds/41002
- https://buildkite.com/elastic/integrations/builds/41120

### PR DESCRIPTION
**Pull request by ritalwar**
## Proposed commit message

- `packages/ess_billing/data_stream/billing`: guard against null API fields

The CEL program used size() on body.instances after checking has(), but has() is also true
for JSON null values. This can cause runtime errors when evaluating
size(null).

- Add explicit != null checks to all has()+size() patterns.

- Also update test data to return null instead of [] for one rule to
exercise the null-handling path in system tests.

- Fix request tracer file path to use the input name

The tracer log directory must match the input type name for logs to be
written to the correct location. Updated it to use "cel" (matching
the input type) instead of the incorrect custom directory names.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://www.elastic.co/docs/extend/integrations/tips-for-building) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).
- [ ] I have verified that any added dashboard complies with Kibana's [Dashboard good practices](https://docs.elastic.dev/ux-guidelines/data-viz/dashboard-good-practices) 

## Author's Checklist

- [ ] 

## Related issues

-

### CODE REVIEW

File: packages/ess_billing/data_stream/billing/_dev/deploy/docker/files/stream-config.yml [hunk 1]
```diff
@@ -34,7 +34,7 @@ rules:
           {{ minify_json `
             {
               "total_ecu": 14.0214,
-              "instances": []
+              "instances": null
```
Comment (andrewkroh) [inline packages/ess_billing/data_stream/billing/_dev/deploy/docker/files/stream-config.yml:37]: There is no test coverage for the `instances: null` case that this PR fixes. The only system test case uses `organization_id: 1111`. And this line is added under org 2222 so it never get used.

https://github.com/elastic/integrations/blob/d62c5cc2b8168bbb14502a5f4aa8de31547e2883/packages/ess_billing/data_stream/billing/_dev/test/system/test-normal-config.yml#L5

**Review by jakubgalecki0 — APPROVED** [review]

File: packages/ess_billing/data_stream/billing/_dev/deploy/docker/files/stream-config.yml (hunk 1 above)
Comment (ritalwar) [reply packages/ess_billing/data_stream/billing/_dev/deploy/docker/files/stream-config.yml:37]: Right, I updated it to align with the new defensive null handling and manually verified the instances: null behavior separately that generates events with fake true, that eventually gets dropped. Since elastic-package currently doesn’t support zero hit_count assertions, no point adding separate config(using org id 2222) for it. WDYT?

**Review by stefans-elastic — APPROVED** [review]
LGTM

File: packages/ess_billing/data_stream/billing/_dev/deploy/docker/files/stream-config.yml (hunk 1 above)
Comment (andrewkroh) [reply packages/ess_billing/data_stream/billing/_dev/deploy/docker/files/stream-config.yml:37]: > and manually verified the instances: null 

Ok, good. I wanted to make sure this line wasn't the only validation.

**Review by andrewkroh — APPROVED** [review]

File: packages/ess_billing/data_stream/billing/agent/stream/cel.yml.hbs [hunk 2]
```diff
@@ -1,7 +1,7 @@
 config_version: 2
 interval: "24h"
 {{#if enable_request_tracer}}
-resource.tracer.filename: "/run/service_logs/http-request-trace-*.ndjson"
+resource.tracer.filename: "../../logs/cel/http-request-trace-*.ndjson"
```
Comment (ishleenk17) [inline packages/ess_billing/data_stream/billing/agent/stream/cel.yml.hbs:4]: Is this change intentional ? 
Has the location changed ?

Comment (ritalwar) [reply packages/ess_billing/data_stream/billing/agent/stream/cel.yml.hbs:4]: Yes, detailed in PR description comment as well.

**Comment by jamesagarside** [comment]
Any chance we can remove customer-architects as a codeowner please, its notifying us all on each update

**Review by ishleenk17 — APPROVED** [review]
